tsweb/tsweb.go

```diff
@@ -114,3 +111,64 @@ func stripPort(hostport string) string {
 	}
 	return net.JoinHostPort(host, "443")
 }
+
+// varzHandler is an HTTP handler to write expvar values into the
+// prometheus export format:
+//
+//   https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md
+//
+// It makes the following assumptions:
+//
+//   * *expvar.Int are counters.
+//   * an *expvar.Map is descended into, joining keys with
+//     underscores. So use underscores as your metric names.
+//   * an expvar named starting with "gauge_" is of type gauge.
+//   * an expvar named starting with "counter_" is of type counter.
+//   * anything else is untyped (for now)
+//   * expvar.Func can return an int or int64 (for now), but if it
+//     doesn't have a type name it's skpiped.
+//
+// This will evolve over time, or perhaps be replaced.
```

Comment:
Note that evolving this requires throwing away data in our tsdb. This is a reason I'm anxious to get it as right as possible out of the gate, because the cost of mistakes is living with them for as long as we want unbroken data.

Reply:
Ack. But I looked at the output and it all seems reasonable.